port/port_posix.cc

```diff
@@ -48,6 +61,19 @@ void CondVar::Wait() {
   PthreadCall("wait", pthread_cond_wait(&cv_, &mu_->mu_));
 }
 
+bool CondVar::Wait(struct timespec* pTimespec) {
+  int result = pthread_cond_timedwait(&cv_, &mu_->mu_, pTimespec);
+  if (0 == result) {
+    return true; // condition variable signaled
```

Comment:
Did I ever explain why I really, really hate multiple "return" statements in a function?